src/Stratis.Features.Diagnostic/Utils/ReflectionExtensionsion.cs

```diff
@@ -54,5 +54,18 @@ public static T GetPrivateFieldValue<T>(this object obj, string fieldName)
             FieldInfo field = obj.GetType().GetField(fieldName, BindingFlags.NonPublic | BindingFlags.Instance);
             return (T)field?.GetValue(obj);
         }
+
+        /// <summary>
+        /// Using reflection, calls a private method passing along the supplied parameters.
+        /// </summary>
+        /// <param name="obj">Object from where the Property Value is set</param>
+        /// <param name="methodName">Name of the method.</param>
+        /// <param name="parameters">Parameters for the method.</param>
+        /// <returns>The method return value.</returns>
+        public static object CallPrivateMethod(this object obj, string methodName, params object[] parameters)
```

Comment:
If we want to keep that method, consider adding this moethod to `Stratis.Bitcoin`.

Reply:
Done.